### PR TITLE
Adding explicit length of git revision in Makefile and E2E Can't Allocate test 

### PR DESCRIPTION
**Pull request by aLekSer**
Output of git rev-parse short command could be different for different environments. For example on my laptop I got 8 chars which leads to errors on deployment to test-cluster using make install.
Namely:
```
ImagePullBackOff     agones-controller: gcr.io/agones-images/agones-controller:0.7.0-5e2ad5ee 
```
which should be gcr.io/agones-images/agones-controller:0.7.0-5e2ad5e.
Also adding separate test for "Can't allocate on fully sized fleet".

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -35,7 +35,7 @@ release_registry = gcr.io/agones-images
 #
 
 # Version defaults to the short hash of the latest commit
-VERSION ?= $(base_version)-$(shell git rev-parse --short HEAD)
```
Comment (markmandel) [inline build/Makefile]: This is good catch! Should this be in a separate PR? Happy to have it in 1, but we usually squash down to a single commit for each PR.

File: test/e2e/fleet_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 	e2e "agones.dev/agones/test/e2e/framework"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
-	"k8s.io/api/apps/v1"
+	v1 "k8s.io/api/apps/v1"
```
Comment (markmandel) [inline test/e2e/fleet_test.go]: Minor thing - either we should drop the `v1` import alias (as it's the same thing), or let's make it `appsv1` (which I'm pretty sure what we call it elsewhere)

File: test/e2e/fleet_test.go [hunk 3]
```diff
@@ -318,14 +366,16 @@ func scaleFleet(f *v1alpha1.Fleet, scale int32) (*v1alpha1.Fleet, error) {
 	return framework.AgonesClient.StableV1alpha1().Fleets(defaultNs).Patch(f.ObjectMeta.Name, types.JSONPatchType, []byte(patch))
 }
 
+const replicasCount = 3
```
Comment (markmandel) [inline test/e2e/fleet_test.go]: Let's move this up to the top with the rest of the consts - that's the usual go way.

File: test/e2e/fleet_test.go (hunk 2 above)
Comment (aLekSer) [reply test/e2e/fleet_test.go]: This is VS Code thing. It was added automatically. Agree that appsv1 feets more natural.
